gcc/cp/contracts.cc

```diff
@@ -1305,6 +1305,11 @@ match_deferred_contracts (tree decl)
 static GTY(()) hash_map<tree, tree> *decl_pre_fn;
 static GTY(()) hash_map<tree, tree> *decl_post_fn;
 
+/* tree that holds the pseude source location type */
+static GTY(()) tree pseudo_source_location_impl_type;
```

Comment:
(not a big deal but) I'm not entirely sure why Andrew chose to label things "pseudo" - to me that indicates it's not necessarily the same as a "real" one..   We could just s/pseudo/constracts/ if the motivation is to avoid name clashes.

Reply:
left Andrew's code as is, replaced new uses of `pseudo` with `contracts`